app/views/tracks.js

```diff
@@ -1,38 +1,49 @@
 import Mn from 'backbone.marionette';
-import Tracks from '../collections/tracks';
 import TrackView from './track';
+import application from '../application';
 
-const TracksView = Mn.CollectionView.extend({  
-    el: '#app-hook',
-    tagName: 'ul',
+
+const TracksView = Mn.CompositeView.extend({
+
+    template: require('./templates/tracks'),
+
+    childViewContainer: '#track-list',
 
     childView: TrackView,
-    events: {
-        'click a': 'play',
-        'click .delete': 'delete'
+
+    onRender() {
+        this.setCurrentTrack();
+        this.listenTo(
+            application.appState,
+            'change:currentTrack',
+            this.setCurrentTrack
+        );
     },
-    play: function (e) {
-        const id = e.currentTarget.dataset.id;
-        const item = this.collection.get(id);
-        item.getStreamURL(playAudio);
+
+    onDestroy() {
+        try {
+            this.stopListening(); // throw weird errors
```

Comment:
I don't understand what you're trying to stop to listen to here. Can you explain?